<!-- location_saver_hsr.launch -->
<launch>

    <!--  ファイル名を指定するなら下記のargを変更すること -->
    <arg name="map_file" default="$(find robocup_simopl_handyman)/map/LayoutD/LayoutD_movebase.yaml"/>
    <arg name="initial_pose_x" default="0.0"/>
    <arg name="initial_pose_y" default="0.0"/>
    <arg name="initial_pose_a" default="0.0"/>
    <arg name="custom_param_file" default="$(find sobit_navigation)/param/wrs_hsr/costmap_common_params.yaml"/>

    <!-- Map server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

    <!-- AMCL -->
    <include file="$(find sobit_navigation)/launch/includes_wrs_hsr/amcl.launch.xml">
        <arg name="initial_pose_x" value="$(arg initial_pose_x)"/>
        <arg name="initial_pose_y" value="$(arg initial_pose_y)"/>
        <arg name="initial_pose_a" value="$(arg initial_pose_a)"/>
    </include>

    <!-- Move_Base -->
    <include file="$(find turtlebot_navigation)/launch/includes/move_base.launch.xml">
        <arg name="custom_param_file" value="$(arg custom_param_file)"/>
    </include>

    <!--location_saver-->
    <node name="location_saver" pkg="sobit_navigation" type="location_saver.py" output="screen" launch-prefix="gnome-terminal -e" />
    
    <!--waypoint_drawer-->
    <node name ="wp_marker" pkg="sobit_navigation" type="waypoint_drawer.py" output="screen"/>

	<!-- rviz -->
    <include file="$(find turtlebot_rviz_launchers)/launch/view_navigation.launch"/><!--for turtlebot-->


  <group ns="non_obj_broadcaster">
    <param name="map_frame_name" type="string" value="map"/>
    <param name="destination_json_path" type="string" value="$(find robocup_simopl_handyman)/map/LayoutD/LayoutD_destination.json"/>
    <node name="destination_broadcaster_node" pkg="robocup_simopl_handyman" type="destination_broadcaster"/>
  </group>

</launch>
